File: CPP/Easy/Determine-if-String-Halves-Are-Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        int firstCount = 0;
        int secondCount = 0;
        
        for (int i = 0; i<s.length()/2; i++) {
            if (isVowel(s[i]) == true ) {
                firstCount++;
            }
        }
        
        for (int i=s.length()/2; i<s.length(); i++) {
            if (isVowel(s[i]) == true ) {
                secondCount++;
            }
        }
        return (firstCount == secondCount); 
        
    }
    
    bool isVowel(char s) {
       if (s == 'a' || s == 'e' || s == 'i' || s == 'o' || s == 'u') {
           return true;
       } 
       if (s == 'A' || s == 'E' || s == 'I' || s == 'O' || s == 'U') {
           return true;
       }
        return false;
    }
};

File: Swift/Easy/Determine-if-String-Halves-Are-Alike.swift
class Solution {
    func halvesAreAlike(_ s: String) -> Bool {
        let string = Array(s) //Easier to fetch char via indexing 
        let vowels: [String] = ["A", "E", "I", "O", "U", "a", "e", "i", "o", "u"]
        var front = 0
        var end = string.count - 1
        var firstVolCount = 0
        var secondVolCount = 0
        
        while(front<end) {
            if vowels.contains(String(string[front])) {
                firstVolCount += 1
            }
            
            if vowels.contains(String(string[end])) {
                secondVolCount += 1
            }
            
            front+=1 
            end -= 1
            
        }
        return firstVolCount == secondVolCount
        
    }

}

